src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -154,6 +154,93 @@ public static void MergeAttributes(this TagHelperOutput tagHelperOutput, TagBuil
             }
         }
 
+        /// <summary>
+        /// Adds the given <paramref name="classValue"/> to the <paramref name="tagHelperOutput"/>'s
+        /// <see cref="TagHelperOutput.Attributes"/>.
+        /// </summary>
+        /// <param name="tagHelperOutput">The <see cref="TagHelperOutput"/> this method extends.</param>
+        /// <param name="classValue">The class value to add.</param>
+        public static void AddClass(
+            this TagHelperOutput tagHelperOutput,
+            string classValue)
+        {
+            if (tagHelperOutput == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelperOutput));
+            }
+
+            if (!tagHelperOutput.Attributes.ContainsName("class"))
+            {
+                tagHelperOutput.Attributes.Add("class", classValue);
+            }
+            else
+            {
+                var classAttributeValue = tagHelperOutput.Attributes["class"].Value.ToString();
+                if (classAttributeValue.Contains(" "))
+                {
+                    var arrayOfClasses = classAttributeValue.Split(' ');
+                    if (arrayOfClasses.Contains(classValue))
+                    {
+                        return;
+                    }
+                }
+
+                if (!classAttributeValue.Equals(classValue))
+                {
+                    tagHelperOutput.Attributes.SetAttribute("class",
```

Comment:
With the following attribute:
```
@{
    var content = "<hello>";
}
<something class="@content" />
```

You want to make sure the `<hello>` result is raw because the value of the attribute will have it as `&lt;hello&gt;` (since it's in an implicit expression). If it wasn't in a raw string you'd double encode the ampersands.

Also, other TagHelpers have the opportunity to mutate the "class" attribute as well; therefore, your TagHelper has to be very flexible with its changing of that attribute.